x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/capacity/AutoscalingDeciderService.java

```diff
@@ -52,6 +45,18 @@
      */
     List<DiscoveryNodeRole> roles();
 
+    /**
+     * Whether or not the decider applies to a policy that specifies an empty set of roles.
+     *
+     * The default implementation is false, as it is expected that most deciders will apply to specific roles. The application of a policy
+     * that specifies an empty set of roles is useful for testing.
```

Comment:
nit: the purpose of the empty roles was not really testing, rather to match a tier of nodes having no roles whatsoever. I think we can eventually remove this case if we ever introduce the explicit coordinating role(s).

```suggestion
     * that specifies an empty set of roles is useful for a tier of coordinating-only nodes.
```

Reply:
I see, I made the current statement, which I still think holds, because of the existence of the fixed autoscaling decider?

Reply:
Not sure about that. If we were to add a search coordinator decider, it would need to apply to all data roles plus the empty set of roles (plus maybe a few more).

But feel free to just leave it if you prefer, is just a nit.